### ISSUE TITLE
Using the Swift 2.0 branch

### ISSUE THREAD

**Issue by fancy42** [CocoaLumberjack/CocoaLumberjack#588]
When using the Swift 2.0 branch I had trouble using 
DDLogDebug, DDLogWarn, etc...

It took me the following answer to realize I needed to copy the CocoaLumberjack.swift file manually to my XCode Project:
http://stackoverflow.com/a/28195896/1388068

FYI here is my Pod file:
use_frameworks!
target ‘MyTarget’ do
pod 'CocoaLumberjack', :git => 'https://github.com/CocoaLumberjack/CocoaLumberjack.git', :branch => 'swift_2.0'
end

If this is the correct way to go, maybe this could be mentioned in the documentation at some point?

**Comment by rivera-ernesto** [comment]
I think everything is quite beta for that branch yet, including CocoaPods.

In the mean time keeping this issue visible may help some people, thanks.

**Comment by ArtSabintsev** [comment]
Just spoke with @neonichu about this exact issue:

Here's how you get the Swift file in there:

`pod 'CocoaLumberjack/Swift', :git => 'https://github.com/CocoaLumberjack/CocoaLumberjack.git', :branch => 'swift_2.0'`

The reason we have to do that is because `default_subspecs` doesn't declare the Swift Subspace, which is why it doesn't import the Swift file (see this line https://github.com/CocoaLumberjack/CocoaLumberjack/blob/swift_2.0/CocoaLumberjack.podspec#L28)

**Comment by dudeinthemirror** [comment]
Hi Guys, thanks for your awesome work. Now that Xcode 7 GM seed is out, what are your plans for merging 'swift_2.0' branch into master ? Also, if that won't happen in the immediate future, could you please create a beta pre-release off of the 'swift_2.0' branch, so we can publish iOS frameworks which have Cocoalumberjack as dependency ? (the podspec doesn't support take :git ... :branch constructs ( see: https://github.com/CocoaPods/CocoaPods/issues/4127 )

**Comment by rivera-ernesto** [comment]
I think we could do that, keep the Xcode 6.4/Swift 1.2 2.0.x releases and make a 2.1 release for Xcode 7/Swift 2.0. @bpoplauschi could you take a look?

**Comment by ArtSabintsev** [comment]
Mind leaving a note here when you do make a 2.1 release as I'm currently pointing to this branch on a few projects - thanks!

**Comment by PSchu** [comment]
@rivera-ernesto Just wanted to ask if you plan to do a release Version for Xcode7, since today iOS9 will be released to the public?

**Comment by rivera-ernesto** [comment]
I think that for some time we will need to keep both branches as Xcode 7 is available but not _required_ yet so we will still have users using 6.3.

As for releasing it, I am not actively using the project at the moment so I don't feel confident about doing the releases myself, so maybe @bpoplauschi or another collaborator would be better.

**Comment by appsforall** [comment]
Here's another vote to get a release out :-) We are building with Xcode 7 now, but we keep running into this issue too. Thanks!

**Comment by rivera-ernesto** [comment]
@MaddTheSane could probably make a beta 2.1.x release with Swift 2/Xcode 7 support.

**Comment by erichoracek** [comment]
It would be really awesome to get a precompiled binary uploaded in the GitHub release when you're ready to push it out. This would speed up build times of those of us that use Carthage to import CocoaLumberjack. See https://github.com/carthage/carthage#archive-prebuilt-frameworks-into-one-zip-file for details.

**Comment by guidomb** [comment]
Yeah that is true the precompiled binary would be reallu helpfull. Even more in slow envs like travis ci.

**Comment by rivera-ernesto** [comment]
I guess anyone could pull request the binaries.

**Comment by guidomb** [comment]
I am not sure if you can PR a binary file because those belong to the release not to the git repo. The person that can edit a release is the one that should add the binaries. All it has to do is execute `carthage archive CocoaLumberjack` and add the zip file to the release.

**Comment by rivera-ernesto** [comment]
Right. We could still add collaborators to the project.

**Comment by guidomb** [comment]
I have no problem adding the precompiled framework once the release is done but I feel that somebody with previuos experience as a collaborator should cut the release.

**Comment by rivera-ernesto** [comment]
Yes, definitely. And as I said before a collaborator that is actively using the library.

**Comment by dudeinthemirror** [comment]
Guys, any luck creating the beta pre-release off of the 'swift_2.0' branch ? Thanks

**Comment by timbodeit** [comment]
Is there anything standing in the way of a 2.1.0-beta release? Any specific issues that one could get involved in fixing?
Back in August, I created my own Swift 2 beta "release" on a fork, that we have been using since. Because we rely on CocoaLumberjack in multiple of our internal pods, simply specifying `:head` or a git url (like you could in a Podfile) is not an option. 
An official beta release would help us out a lot.

**Comment by bpoplauschi** [comment]
Got it, we will make a beta release asap with the Swift 2.0 support.

**Comment by rivera-ernesto** [comment]
@timbodeit You could pull request your beta release fork. Take a look at https://github.com/CocoaLumberjack/CocoaLumberjack/issues/592#issuecomment-146845968.

**Comment by timbodeit** [comment]
@rivera-ernesto On my current fork, I didn't spend any care on the `Lumberjack.xcodeproj` file as this was not relevant to my Cocoapods installation.
I'm currently redoing the merge with the uptodate swift_2.0 and master branches.

**Comment by jpshelley** [comment]
@timbodeit any progress updates?

**Comment by bpoplauschi** [comment]
@timbodeit merged his changes into our swift branch and we will merge those into master very soon.

**Comment by bpoplauschi** [comment]
I have just released a Beta 2.1.0 version via CocoaPods including the Swift 2.0 support (also available in the latest master). Please test it and tell us if it works as expected.

**Comment by rivera-ernesto** [comment]
Now that master requires Xcode 7, maybe we should keep a Xcode 6.x branch for 2.0.x maintenance releases?

**Comment by bpoplauschi** [comment]
@rivera-ernesto I just realised today that the 2.0.2 also required Xcode 7 because of 5627dff. The issue was I tried to create the release from 04672bd but because I had Xcode 7 on my machine, it would not pass `pod lib lint`.
To fix this, I have created a new release, 2.0.3 that brings back the Xcode 6 compabitility. Also, I created a [Xcode 6 branch](https://github.com/CocoaLumberjack/CocoaLumberjack/tree/Xcode_6) if we need more releases on this line.
